processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -747,15 +815,36 @@ func (c *WatchClient) extractNodeAttributes(node *api_v1.Node) map[string]string
 	return tags
 }
 
+func (c *WatchClient) extractDeploymentAttributes(d *apps_v1.Deployment) map[string]string {
+	tags := map[string]string{}
+
+	for _, r := range c.Rules.Labels {
+		r.extractFromDeploymentMetadata(d.Labels, tags, "k8s.deployment.label.%s")
```

Comment:
Probably make those constants with a comment linking to SemConv?

(I don't know if we'll be able to use the prefix directly from the generated code though)